packages/api/src/models/security/authentication/gdb-token-auth-provider.ts

```diff
@@ -0,0 +1,47 @@
+import {AuthenticatedUser} from '../authenticated-user';
+import {AuthStrategy} from './auth-strategy';
+import {AuthStrategyType} from './auth-strategy-type';
+import {MapperProvider, service} from '../../../providers';
+import {AuthenticatedUserMapper, AuthenticationStorageService, SecurityContextService} from '../../../services/security';
+import {SecurityRestService} from '../../../services/security/security-rest.service';
+
+type LoginData = {
+  username: string;
+  password: string;
+}
+
+export class GdbTokenAuthProvider implements AuthStrategy {
+  private readonly securityRestService = service(SecurityRestService);
+  private readonly authStorageService = service(AuthenticationStorageService);
+  private readonly securityContextService = service(SecurityContextService);
+
+  type = AuthStrategyType.GDB_TOKEN;
+
+  initialize(): Promise<unknown> {
+    return Promise.resolve();
+  }
+
+  async login(loginData: LoginData): Promise<AuthenticatedUser> {
+    const {username, password} = loginData;
+    const response = await this.securityRestService.login(username, password);
+    const authHeader = response.headers.get('authorization');
+    if (authHeader) {
+      this.authStorageService.setAuthToken(authHeader);
+    }
+    const responseData = await response.json();
```

Comment:
What if this blows up? You'd have the token set in the storage, but no user would be configured in the app.